src/data/air_composition.F90

```diff
@@ -639,27 +689,33 @@ end subroutine get_cp_dry_2hd
    !
    !***************************************************************************
    !
-   subroutine get_cp_1hd(tracer, inv_cp, cp, dp_dry, active_species_idx_dycore)
+   subroutine get_cp_1hd(tracer, inv_cp, cp, factor, active_species_idx_dycore, cpdry)
       use cam_abortutils,  only: endrun
       use string_utils,    only: to_str
 
       ! Dummy arguments
       ! tracer: Tracer array
+      !
+      ! factor not present then tracer must be dry mixing ratio
+      ! if factor present tracer*factor must be dry mixing ratio
+      !
       real(kind_phys),           intent(in)  :: tracer(:,:,:)
-      real(kind_phys), optional, intent(in)  :: dp_dry(:,:)
       ! inv_cp: output inverse cp instead of cp
       logical,                   intent(in)  :: inv_cp
       real(kind_phys),           intent(out) :: cp(:,:)
+      ! dp: if provided then tracer is mass not mixing ratio
```

Comment:
Thanks for noticing this. Looks like the line of code below got out of sync with the comment (it is the same in CAM). @PeterHjortLauritzen originally introduced this early last year - could you please confirm this is the intent of the code? Perhaps it was originally designed to convert mass? Although I recognize it might be hard to remember something from almost two years ago...

Reply:
Factor is used to convert ```tracer``` to dry mixing ratio. So if tracer is mass then one can divide by dpdry (factor=dpdry) to get dry mixing ratio. If tracer is moist mixing ratio then ```factor``` is used to convert wet to dry mixing ratio (dpmoist/dpdry).

I guess a more accurate comment would be:

factor to convert ```tracer``` to dry mixing ratio.

Reply:
Thanks, comment updated!